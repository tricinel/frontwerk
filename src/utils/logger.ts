/* eslint @typescript-eslint/unbound-method: "off" */

import consola from 'consola';
import { compose, join } from 'ramda';

const argsToArray = (...args: string[]): string[] => [...args];
const formatMessage: (...args: string[]) => string = compose(
  join('\n'),
  argsToArray
);

const success: (...args: string[]) => void = compose(
  consola.success,
  formatMessage
);
const error: (...args: string[]) => void = compose(
  consola.error,
  formatMessage
);
const warning: (...args: string[]) => void = compose(
  consola.warn,
  formatMessage
);
const info: (...args: string[]) => void = compose(
  consola.info,
  formatMessage
);
const start: (...args: string[]) => void = compose(
  consola.start,
  formatMessage
);

export { formatMessage, success, error, warning, info, start };
