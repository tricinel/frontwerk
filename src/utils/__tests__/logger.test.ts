import { formatMessage } from '../logger';

test('Create the correct message', () => {
  const messages = [
    'Message title!',
    'Message body goes here.',
    'Some other message on another line.'
  ];

  expect(formatMessage(...messages)).toMatchSnapshot();
});
