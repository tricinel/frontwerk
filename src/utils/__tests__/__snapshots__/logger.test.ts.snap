// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create the correct message 1`] = `
"Message title!
Message body goes here.
Some other message on another line."
`;
